### PR TITLE
[New] Add jquery-typeahead w/ git auto-update

### PR DESCRIPTION
**Pull request by extend1994**
Pull request for issue: #12214

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/running-coder/jquery-typeahead
 * Official website (optional, not the repository): http://www.runningcoder.org/jquerytypeahead/
 * NPM package url (optional): https://www.npmjs.com/package/jquery-typeahead
 * License and its reference: [MIT](https://github.com/running-coder/jquery-typeahead/blob/master/LICENSE)
 * GitHub / Bitbucket popularity (required): 
   - Count of watchers: 35
   - Count of stars: 383
   - Count of forks: 95
 * NPM download stats (optional):
   - Downloads in the last day: 0
   - Downloads in the last week: 4,638
   - Downloads in the last month: 27,428

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 200 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 800 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by extend1994** [comment]
1. Use git as the source of auto-update because there are more versions.
2. Didn't use auto-update patterns to fetch old versions because it might cause chaos. Instead, I manually add them.

File: ajax/libs/jquery-typeahead/package.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      {
+        "basePath": "dist",
+        "files": [
+          "**/*"
+        ]
+      },
+      {
+        "basePath": "src",
+        "files": [
+          "jquery.typeahead!(*.scss)"
```
Comment (sufuf3) [inline ajax/libs/jquery-typeahead/package.json]: I think we use included mode when `src` as the bathPath.
Like `jquery.typeahead*.+(css|js|map)`
What do you think?
Thanks.

Comment (extend1994) [reply ajax/libs/jquery-typeahead/package.json]: It was my mistake, will update it, thx.

**Comment by extend1994** [comment]
@sufuf3 Both requests fixed. Please help review it, thanks.

**Comment by PeterDaveHello** [comment]
@extend1994 can you explain more details about the chaos? I think using auto-update to fetch and then manually reviewing would be a better way.

**Pull request merged**